### ISSUE TITLE
[java] AccessorMethodGeneration: Method inside static inner class incorrectly reported

### ISSUE THREAD

**Issue by SUPERCILEX** [pmd/pmd#274]
**Rule Set:** design.html#AccessorMethodGeneration

**Code Sample demonstrating the issue:**

```java
public static final Creator<Team> CREATOR = new Creator<Team>() {
    @Override
    public Team createFromParcel(Parcel source) {
        return new Team(source.readString(),
                        source.readString(),
                        source.readString(),
                        source.readString(),
                        source.readString(),
                        source.readString(),
                        getBooleanForInt(source.readInt()), // These are the offending lines
                        getBooleanForInt(source.readInt()), // same
                        getBooleanForInt(source.readInt()), // same
                        source.readLong());
    }

    @Override
    public Team[] newArray(int size) {
        return new Team[size];
    }

    private boolean getBooleanForInt(int value) {
        return value == 1;
    }
};
```

**Running PMD through:** Gradle

**Comment by jsotuyod** [comment]
@SUPERCILEX thanks for the report. The issue is confirmed, I'm working on a fix.

**Comment by jsotuyod** [comment]
PR submitted. Interestingly this pointed me towards another deeper bug with symbol table analysis, which is now also fixed.

**Comment by SUPERCILEX** [comment]
Awesome! Glad to have helped you fix two bugs! 😄
